From fc700aa5430881a88fca00cb36d24d6f1d2e75b3 Mon Sep 17 00:00:00 2001
Message-ID: <fc700aa5430881a88fca00cb36d24d6f1d2e75b3.1738269782.git.maciek.borzecki@gmail.com>
In-Reply-To: <cover.1738269782.git.maciek.borzecki@gmail.com>
References: <cover.1738269782.git.maciek.borzecki@gmail.com>
From: Maciek Borzecki <maciek.borzecki@gmail.com>
Date: Thu, 30 Jan 2025 21:32:12 +0100
Subject: [PATCH 1/2] Allow overriding default OVMF code and vars blobs
 locations

Not all systems package ovmf binary blobs the same way. Provide lazy
defaults and let packagers override as needed.

Signed-off-by: Maciek Borzecki <maciek.borzecki@gmail.com>
---
 mk/010-qemu.mk | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/mk/010-qemu.mk b/mk/010-qemu.mk
index 0d1f38ce77eed5be512b7c8940f1436aec05fe4b..faf43f6982471fb1d21a89dcc640968c9a084bdb 100644
--- a/mk/010-qemu.mk
+++ b/mk/010-qemu.mk
@@ -72,16 +72,23 @@ QEMU_SYSTEM_AARCH64_EFI_CMDLINE = \
 	$${QEMU_NET_OPTION=-netdev user,id=netdev0$${QEMU_NETDEV_USER_EXTRA:-} \
 		-device virtio-net-pci,netdev=netdev0,id=net0}
 
-efi-code.x86_64.img: $(or $(firstword $(wildcard /usr/share/OVMF/OVMF_CODE.fd /usr/share/OVMF/OVMF_CODE_4M.fd)),$(error please install ovmf firmware))
+# default location on most Debian/Ubuntu systems
+QEMU_OVMF_CODE_X86 ?= /usr/share/OVMF/OVMF_CODE.fd /usr/share/OVMF/OVMF_CODE_4M.fd
+QEMU_OVMF_VARS_X86 ?= /usr/share/OVMF/OVMF_VARS.fd /usr/share/OVMF/OVMF_VARS_4M.fd
+
+efi-code.x86_64.img: $(or $(firstword $(wildcard $(QEMU_OVMF_CODE_X86))),$(error please install ovmf firmware))
 	dd if=$< of=$@ conv=notrunc
 
-efi-vars.x86_64.img: $(or $(firstword $(wildcard /usr/share/OVMF/OVMF_VARS.fd /usr/share/OVMF/OVMF_VARS_4M.fd)),$(error please install ovmf firmware))
+efi-vars.x86_64.img: $(or $(firstword $(wildcard $(QEMU_OVMF_VARS_X86))),$(error please install ovmf firmware))
 	dd if=$< of=$@ conv=notrunc
 
-efi-code.aarch64.img: /usr/share/AAVMF/AAVMF_CODE.fd
+QEMU_OVMF_CODE_AARCH64 ?= /usr/share/AAVMF/AAVMF_CODE.fd
+QEMU_OVMF_VARS_AARCH64 ?= /usr/share/AAVMF/AAVMF_VARS.fd
+
+efi-code.aarch64.img: $(QEMU_OVMF_CODE_AARCH64)
 	dd if=$< of=$@ conv=notrunc
 
-efi-vars.aarch64.img: /usr/share/AAVMF/AAVMF_VARS.fd
+efi-vars.aarch64.img: $(QEMU_OVMF_VARS_AARCH64)
 	dd if=$< of=$@ conv=notrunc
 
 # UEFI firmware code and state.
-- 
2.48.1

